[arm64] Assertion failed 'opt != INS_OPTS_1D' shuffling Vector64&lt;nint&gt;/Vector64&lt;nuint&gt; with variable indices

### Description

On arm64, a checked JIT asserts when compiling `Vector64.Shuffle` / `Vector64.ShuffleNative` for a 64-bit element type with **variable** indices:

```
Assertion failed 'opt != INS_OPTS_1D' in
'System.Runtime.Intrinsics.Vector64:ShuffleNative(System.Runtime.Intrinsics.Vector64`1[nuint],System.Runtime.Intrinsics.Vector64`1[nuint]):System.Runtime.Intrinsics.Vector64`1[nuint]'
during 'Generate code' (IL size 8; hash 0x6f28d991; FullOpts)
```

from `src/coreclr/jit/emitarm64.cpp:5394`.

### Root cause

`Compiler::gtNewSimdShuffleVariableNode` scales variable shuffle indices into byte offsets (`gentree.cpp`):

```cpp
// shift all indices to the left by tzcnt(size)
cnsNode = gtNewIconNode(BitOperations::TrailingZeroCount(static_cast<uint64_t>(elementSize)), TYP_INT);
op2     = gtNewSimdHWIntrinsicNode(type, op2, cnsNode, NI_AdvSimd_ShiftLeftLogical, simdBaseType, simdSize);
```

When `simdSize == 8` and `elementSize == 8`, the vector holds a single 64-bit element, i.e. a `1D` arrangement. The *vector* form of the arm64 shift-by-immediate has no valid `1D` encoding (reserved) — that case must use the *scalar* form.

The JIT already knows this. `gtNewSimdBinOpNode` applies exactly the right guard for `GT_LSH`:

```cpp
#elif defined(TARGET_ARM64)
    if ((simdSize == 8) && (genTypeSize(simdBaseType) == 8))
    {
        id = op2->IsCnsIntOrI() ? NI_AdvSimd_ShiftLeftLogicalScalar : NI_AdvSimd_ShiftLogicalScalar;
    }
    else
    {
        id = op2->IsCnsIntOrI() ? NI_AdvSimd_ShiftLeftLogical : NI_AdvSimd_ShiftLogical;
    }
```

The shuffle path constructs the HWINTRINSIC node directly and so never gets that guard. The adjacent code does special-case `(simdSize == 16) && (simdBaseType == TYP_INT)`, but nothing handles the 8/8 case.

### Conditions required to hit it

- **arm64**, checked JIT (the assert is checked-only).
- **A 64-bit element type in a `Vector64`.** `nint` / `nuint` are the only `Vector64` shuffle overloads with one — there are no `long` / `ulong` / `double` overloads — and only on a 64-bit platform.
- **Variable indices.** Constant indices take a different path that never emits the shift.

Because `elementCount == 1` for these types, the index fix-up is degenerate; it is emitted only because the generic path always emits it.

### Impact

Most visible under NativeAOT: ILC compiles CoreLib whole-program and instantiates `ShuffleNative<nuint>` regardless of whether anything calls it, so **any NativeAOT build using a checked arm64 JIT fails**:

```
error MSB3073: The command "…/ilc @…/System.Private.CoreLib.ilc.rsp" exited with code 133.
```

This blocks `src/tests/build.sh nativeaot checked` on an arm64 host entirely.

### CI exposure

Not caught today. `windows-x64 Checked NativeAOT` is the only checked NativeAOT leg and x64 does not go through `emitarm64.cpp`; the arm64 NativeAOT legs are all Release, where the assert is compiled out.

### Repro

Reproduced on osx-arm64 against `2aadf336` and current `main`, on clean full builds (`build.sh -s clr.aot+libs.native+libs.sfx -rc Checked -lc Release`). A minimal managed repro, with the indices kept variable by a non-inlined call:

```csharp
[MethodImpl(MethodImplOptions.NoInlining)]
static Vector64<nuint> ShuffleNoInline(Vector64<nuint> vector, Vector64<nuint> indices)
    => Vector64.Shuffle(vector, indices);

// on a checked arm64 JIT:
ShuffleNoInline(Vector64.Create((nuint)1), Vector64.Create((nuint)0));
```

```
Assert failure: Assertion failed 'opt != INS_OPTS_1D' in
'…Vector64Tests:ShuffleNoInline(Vector64`1[nuint],Vector64`1[nuint]):Vector64`1[nuint]'
during 'Generate code' (IL size 8; hash 0xf1ceddf2; Tier0)
exit code 134 — SIGABRT
```

### Fix

Mirror the guard already used by `gtNewSimdBinOpNode` and select `NI_AdvSimd_ShiftLeftLogicalScalar` when `(simdSize == 8) && (elementSize == 8)`.

> [!NOTE]
> This issue was prepared with GitHub Copilot assistance and reviewed by the submitting developer.

--- a/src/coreclr/jit/gentree.cpp
+++ b/src/coreclr/jit/gentree.cpp
@@ -29041,8 +29041,12 @@ GenTree* Compiler::gtNewSimdShuffleVariableNode(
         }
 
         // shift all indices to the left by tzcnt(size)
+        // a 64-bit element in a Vector64 is a 1D arrangement, which the vector form of the shift
+        // has no encoding for; that case has to use the scalar form instead.
+        NamedIntrinsic shiftIntrinsic =
+            ((simdSize == 8) && (elementSize == 8)) ? NI_AdvSimd_ShiftLeftLogicalScalar : NI_AdvSimd_ShiftLeftLogical;
         cnsNode = gtNewIconNode(BitOperations::TrailingZeroCount(static_cast<uint64_t>(elementSize)), TYP_INT);
-        op2     = gtNewSimdHWIntrinsicNode(type, op2, cnsNode, NI_AdvSimd_ShiftLeftLogical, simdBaseType, simdSize);
+        op2     = gtNewSimdHWIntrinsicNode(type, op2, cnsNode, shiftIntrinsic, simdBaseType, simdSize);
 
         // VectorTableLookup is only valid on byte/sbyte
         simdBaseType = varTypeIsUnsigned(simdBaseType) ? TYP_UBYTE : TYP_BYTE;

--- a/src/libraries/System.Runtime.Intrinsics/tests/Vectors/Vector64Tests.cs
+++ b/src/libraries/System.Runtime.Intrinsics/tests/Vectors/Vector64Tests.cs
@@ -2436,6 +2436,102 @@ namespace System.Runtime.Intrinsics.Tests.Vectors
             }
         }
 
+        // nint/nuint are the only Vector64 shuffle overloads with a 64-bit element type, and only on
+        // a 64-bit platform. The indices are passed through a non-inlined call so they stay variable,
+        // which is what selects the index fix-up path in the JIT.
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private static Vector64<nint> ShuffleNativeNoInline(Vector64<nint> vector, Vector64<nint> indices) => Vector64.ShuffleNative(vector, indices);
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private static Vector64<nuint> ShuffleNativeNoInline(Vector64<nuint> vector, Vector64<nuint> indices) => Vector64.ShuffleNative(vector, indices);
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private static Vector64<nint> ShuffleNoInline(Vector64<nint> vector, Vector64<nint> indices) => Vector64.Shuffle(vector, indices);
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private static Vector64<nuint> ShuffleNoInline(Vector64<nuint> vector, Vector64<nuint> indices) => Vector64.Shuffle(vector, indices);
+
+        [Fact]
+        public void Vector64NIntShuffleWithVariableIndicesTest()
+        {
+            Vector64<nint> vector = Vector64<nint>.Zero;
+            Vector64<nint> indices = Vector64<nint>.Zero;
+
+            for (int index = 0; index < Vector64<nint>.Count; index++)
+            {
+                vector = vector.WithElement(index, (nint)(index + 1));
+                indices = indices.WithElement(index, (nint)(Vector64<nint>.Count - index - 1));
+            }
+
+            Vector64<nint> result = ShuffleNoInline(vector, indices);
+
+            for (int index = 0; index < Vector64<nint>.Count; index++)
+            {
+                Assert.Equal((nint)(Vector64<nint>.Count - index), result.GetElement(index));
+            }
+        }
+
+        [Fact]
+        public void Vector64NUIntShuffleWithVariableIndicesTest()
+        {
+            Vector64<nuint> vector = Vector64<nuint>.Zero;
+            Vector64<nuint> indices = Vector64<nuint>.Zero;
+
+            for (int index = 0; index < Vector64<nuint>.Count; index++)
+            {
+                vector = vector.WithElement(index, (nuint)(index + 1));
+                indices = indices.WithElement(index, (nuint)(Vector64<nuint>.Count - index - 1));
+            }
+
+            Vector64<nuint> result = ShuffleNoInline(vector, indices);
+
+            for (int index = 0; index < Vector64<nuint>.Count; index++)
+            {
+                Assert.Equal((nuint)(Vector64<nuint>.Count - index), result.GetElement(index));
+            }
+        }
+
+        [Fact]
+        public void Vector64NIntShuffleNativeWithVariableIndicesTest()
+        {
+            Vector64<nint> vector = Vector64<nint>.Zero;
+            Vector64<nint> indices = Vector64<nint>.Zero;
+
+            for (int index = 0; index < Vector64<nint>.Count; index++)
+            {
+                vector = vector.WithElement(index, (nint)(index + 1));
+                indices = indices.WithElement(index, (nint)(Vector64<nint>.Count - index - 1));
+            }
+
+            Vector64<nint> result = ShuffleNativeNoInline(vector, indices);
+
+            for (int index = 0; index < Vector64<nint>.Count; index++)
+            {
+                Assert.Equal((nint)(Vector64<nint>.Count - index), result.GetElement(index));
+            }
+        }
+
+        [Fact]
+        public void Vector64NUIntShuffleNativeWithVariableIndicesTest()
+        {
+            Vector64<nuint> vector = Vector64<nuint>.Zero;
+            Vector64<nuint> indices = Vector64<nuint>.Zero;
+
+            for (int index = 0; index < Vector64<nuint>.Count; index++)
+            {
+                vector = vector.WithElement(index, (nuint)(index + 1));
+                indices = indices.WithElement(index, (nuint)(Vector64<nuint>.Count - index - 1));
+            }
+
+            Vector64<nuint> result = ShuffleNativeNoInline(vector, indices);
+
+            for (int index = 0; index < Vector64<nuint>.Count; index++)
+            {
+                Assert.Equal((nuint)(Vector64<nuint>.Count - index), result.GetElement(index));
+            }
+        }
+
         [Fact]
         public void Vector64SByteShuffleNativeOneInputTest()
         {

--- a/src/libraries/System.Runtime.Intrinsics/tests/Vectors/Vector64Tests.cs
+++ b/src/libraries/System.Runtime.Intrinsics/tests/Vectors/Vector64Tests.cs
@@ -2436,9 +2436,10 @@ namespace System.Runtime.Intrinsics.Tests.Vectors
             }
         }
 
-        // nint/nuint are the only Vector64 shuffle overloads with a 64-bit element type, and only on
-        // a 64-bit platform. The indices are passed through a non-inlined call so they stay variable,
-        // which is what selects the index fix-up path in the JIT.
+        // nint/nuint are the only Vector64 shuffle overloads with a pointer-sized element, so on a
+        // 64-bit platform they are the only way to reach a 64-bit element type (a 1D arrangement).
+        // The indices are passed through a non-inlined call so they stay variable, which is what
+        // selects the index fix-up path in the JIT.
 
         [MethodImpl(MethodImplOptions.NoInlining)]
         private static Vector64<nint> ShuffleNativeNoInline(Vector64<nint> vector, Vector64<nint> indices) => Vector64.ShuffleNative(vector, indices);

